gameplay-controller.js

```diff
@@ -35,11 +33,13 @@ export default class GameplayController {
             kills: 0,
             milestone: 0,
             weapons: [0],
+            score: 0,
         };
         this.controllerState = {
```

Comment:
**praise**: Good use of `Math.random()` for generating randomness, which improves the unpredictability of weapon spawning.